app.js

```diff
@@ -1,9 +1,26 @@
 const express = require("express");
 const app = express()
-const { getTopics, getEndPoints } = require('./controllers/topics-controller')
+const { getTopics } = require('./controllers/topics-controller')
+const { getArticleById } = require('./controllers/articles-controller')
+const { getEndPoints } = require('./controllers/endpoints-controller')
 
 app.get("/api/topics", getTopics);
-console.log(getEndPoints)
+
+app.get("/api/articles/:article_id", getArticleById);
+
 app.get("/api", getEndPoints);
 
+app.use((err, req, res, next) => {
+    if (err.msg === 'Not found'){
+      res.status(404).send({msg: err.msg})
+    }
+    else next(err)
+  })
```

Comment:
Mostly well setup, maybe extend this by just checking if the `err` object contains a `msg` property at all rather than checking if the value is exactly `Not found`, that way if a `msg` property exists, just send that in the response like you have. These notes should help highlight the point: https://notes.northcoders.com/courses/js-back-end/error-handling#custom-errors